src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type FigureShape = 'triangle' | 'circle' | 'rectangle';
+type FigureColor = 'red' | 'green' | 'blue';
+type AreaCallback = () => number;
 
+export interface Figure {
+  shape: FigureShape,
+  color: FigureColor,
+  getArea: AreaCallback,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: FigureShape = 'triangle';
+
+  constructor(
+    public color: FigureColor,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a * b * c <= 0) {
+      throw new Error('You have entered incorrect data');
```

Comment:
Give more descriptive error messages. Why data is incorrect -> because user entered negative numbers